### ISSUE TITLE
Factions.

### ISSUE THREAD

**Issue by Karlinator** [Karlinator/roll20-character-sheets#22]

**Comment by PanzerUXI** [comment]
**## Base Functionality**

**Faction Actor**
- Hit Points
- Attributes - Force, Cunning, Wealth Rating: 1 to 8, determines which assets can be purchased.
- FacCreds
- Goals & XP: Current XP and Goal assigned, with description of mechanic.
- Tags: With description and associated bonuses
- Homeworld: or current Base of Operations
- Faction Asset List

**Faction Asset**
- Type: Force, Cunning, Wealth (tied to Faction Main rating, 3 Force = 3 Force Assets max)
- Hit Points
- Cost
- Attack
- Counterattack
- Type
- Current Location
- Required TL to purchase/own

** Faction Actions **
Roll Template reference?

**Comment by Karlinator** [comment]
There also needs to be a place to track Bases of Influence, as I think those don't really follow the same rules as the rest of the assets. I'd maybe put that in its own area.

Also I think the whole faction sheet needs lots of areas to make notes in for things like descriptions of the asset (especially for bases of influence).

**Comment by joesinghaus** [comment]
Do we want a model of 1 sheet per faction, or 1 faction sheet containing all the factions? It seems we are tending towards the former?

**Comment by PanzerUXI** [comment]
I agree that 1 sheet per faction makes the most sense.

**Comment by Karlinator** [comment]
The downside with one sheet per faction is that it makes accessing data from other factions for things like rolling much harder. Other than that we've always ben leaning heavily towards one actor per sheet so I think that's what we're going with.

Also, in general, there are a few features that I've noticed in the google sheet commonly used that would be hard to replicate. Things like importing the sector map from Sectors Without Number and giving location dropdowns etc for assets from that. Though, could we perhaps do that by having the user paste the sector map JSON export into a field somewhere and then extracting the data from that in the worker? We can do all sorts of stuff with JSON I believe, so that should work. It would have to make unto each sheet, so sheet default setting or pasting in for every faction sheet.

The google sheet also has a convenient list of the faction turn order where you can fill in what each faction did in their turn to keep track of it all. This could be solved with a repeating section on each sheet where you fill in turn number (or date or something) and actions taken.

**Comment by PanzerUXI** [comment]
Thats a solid idea.

**Comment by Karlinator** [comment]
I'd like to propose that we officially downprioritise this feature, as there exist both a bunch of spreadsheets that handle this pretty well, as well as Sectors Without Number now having a faction tracker. These are not bound by the various limitations of Roll20, and are, in my opinion, in many ways better suited for most of the faction management.

**Comment by jbogrand** [comment]
Those tools are not roll20 and it would be good to have the sector within the table top rather than going to another source to do those parts of the game.  Also support of an outside tool to inconsistent into the factions would be inconsistent with how it is done throughout roll20s character sheets.   Someone new to the table top but familiar to the game shouldn't be looking to go to yet another tool but instead thinking to utilize this sheet that already supports other type of actors/agents like ships, npcs and drones.  So there is my 2 cents.

**Comment by Karlinator** [comment]
I agree that it isn't ideal, but honestly the limitations placed on us by Roll20 make it nearly impossible to compete with these tools. Sectors Without Number has gotten *soo good*, I'm afraid I'd be wasting my time trying to make something better/equal within the limitations of Roll20.

Not to mention that a sector map is completely out of the question here. There is no way we're doing that. I'm pretty sure it's impossible, especially without the API, which is a Pro feature. Maybe you can have a "sector map" sheet, but it would be a horribly complicated monstrosity. You can make one on your own using folders and handouts (and I have done so in the past), but if you want a decent sector creator/manager you'll already be using Sectors Without Number.

Really, I think there's a reasonable divide that things that happen is session, things players use, should be supported in the character sheet as much as possible, and things managed outside sessions, things pertaining to the world at large that the GM manages in his prep, can be outsourced to other tools. Every GM will have their own system of keeping notes anyway, so using some outside system for faction isn't far-fetched.

As said, the idea of integrating factions isn't completely dead, but a quick look at the issue tracker here will reveal that there are much more important improvements lined up that affect players in session, not just GMs between.
